terminal.go

```diff
@@ -92,10 +107,32 @@ func (t *Terminal) History() []string {
 }
 
 // NewHistory creates/resets the history to a new one with the given capacity.
+// need + 1 to fit "pending" command.
 func (t *Terminal) NewHistory(capacity int) {
+	if capacity < 0 {
+		log.Errf("Invalid history capacity %d, ignoring", capacity)
+		return
+	}
+	t.capacity = capacity
 	t.term.NewHistory(capacity)
 }
 
+// SetAutoHistory enables/disables auto history (default is enabled).
+func (t *Terminal) SetAutoHistory(enabled bool) {
+	t.autoHistory = enabled
+	t.term.AutoHistory(enabled)
+}
+
+// AutoHistory returns the current auto history setting.
+func (t *Terminal) AutoHistory() bool {
+	return t.autoHistory
```

Comment:
I know you forked golang/term in fortio, but I find strange that you now have two flags about autoHistory, one in fortio/term and one in fortio/terminal

BTW, maybe renaming fortio/term to something like fortio/golang-term-fork would make it easier to understand why fortio has now two "term" packages

Reply:
as mentioned, this one (`terminal`) is standalone and fully encapsulate what happens to be an implementation detail

this being said, others that for instance don't want fortio cli/logger/etc... might use the lower level `term` directly if they also don't want to wait for PRs to merge upstream (which may or may not ever happen)